### PR TITLE
Analytics branch

### PR DESCRIPTION
**Pull request by gaholt**

### CODE REVIEW

File: EDSAnalytics/Coverity.proj [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (dcbroad3) [inline EDSAnalytics/Coverity.proj]: If you copy the [analysis template](https://github.com/osisoft/OSI-Samples-OMF/blob/master/miscellaneous/build_templates/analysis.yml) used by the other projects into this repo and reference that, it should run Coverity (as well as BlackDuck/BinSkim) and this file shouldn't be necessary. (we are in the process of migrating to Polaris anyway)

File: EDSAnalytics/EDSAnalytics.sln [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿
```
Comment (dcbroad3) [inline EDSAnalytics/EDSAnalytics.sln]: Typically we add a `Solution Items` folder to the solution file with common sample files like the README, pipeline, etc, see [here](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/AuthorizationCodeFlow/DotNet/AuthorizationCodeFlow.sln) for an example.

File: EDSAnalytics/EDSTest/EDSAnalytics.csproj [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/EDSAnalytics.csproj]: We have been trying to add StyleCop and FxCop analyzers to our projects, see [this project](https://github.com/osisoft/OSI-Samples-OCS/tree/master/basic_samples/Authentication/ClientCredentialFlow/DotNet) for an example. If you don't have the bandwidth to do it, we can either work on the branch to do it or go back and add that later.

File: EDSAnalytics/EDSTest/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
+
+[![Build Status](https://PLACE_BUILD_STATUS_LINK_HERE)]
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: We can add this if you'd like, but first I think the `azure-pipeline.yml` file needs to be moved up one level in the folder hierarchy.

File: EDSAnalytics/EDSTest/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## Requirements
+
+- .NET Core 3.1 is installed
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: I'd combine the `## Requirements` and `### Prerequisites` sections

File: EDSAnalytics/EDSTest/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
+
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: Add `**Version:**: 1.0.0` between header and build status badge

File: EDSAnalytics/EDSTest/README.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+- Creates the EdsApiAggregatedData stream using SDS
+
+Step 11
+- Data is ingressed from EDS�s standard data aggregate API calls and stored in an object. Since data is encoded using GZIP, some decoding is completed 
+to extract the mean, minimum, maximum, and range. This data is copied into an AggregateData object and sent to the EdsApiAggregatedData stream
+
+Step 12 
+- The Types and Streams created by the application are deleted.
+
+
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: We usually add a common list of links to the bottom of this file, see [this README](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/ClientCredentialFlow/DotNet/README.md) for an example

File: EDSAnalytics/EDSTest/README.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: I think the `EDSTest` and `EDSTestTests` folders should be renamed to `EDSAnalytics` and `EDSAnalyticsTests` respectively

Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: This README file should be moved up one level in the folder hierarchy.

File: EDSAnalytics/EDSTest/azure-pipeline.yml [hunk 9]
```diff
@@ -0,0 +1,72 @@
+trigger:
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/azure-pipeline.yml]: This file should be moved up one level in the folder hierarchy

File: EDSAnalytics/EDSTest/azure-pipeline.yml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        workingDirectory: $(Build.SourcesDirectory)$(projPath)/
+        displayName: 'Build app'
+
+      - script: |
+          dotnet restore
+          dotnet test
+        workingDirectory: $(Build.SourcesDirectory)$(testPath)/
+        displayName: 'Run tests'
+
+  - template: '/miscellaneous/build_templates/analysis.yml'
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/azure-pipeline.yml]: This references a file that has not yet been added to the repository. You can copy it (and the other files in the `/miscellaneous/build_templates/` folder) from [here](https://github.com/osisoft/OSI-Samples-OMF/blob/master/miscellaneous/build_templates/analysis.yml). Without all three files (config/variables/analysis) this build pipeline won't work.

File: EDSAnalytics/EDSTest/azure-pipeline.yml [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    pool:
+      vmImage: $(imageName)
+    steps:
+      - template: '/miscellaneous/build_templates/config.yml'
+        parameters:
+          fileOriginal: $(CredFile)
+          fileLibrary: $(CredFileLibrary)
+          filePath: $(projPath)
+
+      - script: |
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/azure-pipeline.yml]: There's actually a much cleaner way to run/test the app, see [this pipeline](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/ClientCredentialFlow/DotNet/azure-pipeline.yml#L54) for an example

File: EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.3" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
+    <PackageReference Include="xunit" Version="2.4.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
+    <PackageReference Include="coverlet.collector" Version="1.0.1" />
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj]: Is `coverlet` actually used? Looks neat, not something I've used before.

File: EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.3" />
```
Comment (dcbroad3) [inline EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj]: Reminder that if you do add StyleCop and FxCop analyzers, they need to be added to this project as well.

File: EDSAnalytics/EDSTest/README.md (hunk 8 above)
Comment (dcbroad3) [inline EDSAnalytics/EDSTest/README.md]: This sample needs a HISTORY.md file, see [HISTORY.md](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/AuthorizationCodeFlow/DotNet/HISTORY.md) for an example. Right now should just include 1.0.0 / {date}, Initial public release

Comment (gaholt) [reply EDSAnalytics/EDSTest/README.md]: Done.

File: EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj (hunk 12 above)
Comment (gaholt) [reply EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj]: Looks like that was copied forward from old unit tests.

Comment (gaholt) [reply EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj]: Removed.

File: EDSAnalytics/EDSTest/azure-pipeline.yml (hunk 10 above)
Comment (gaholt) [reply EDSAnalytics/EDSTest/azure-pipeline.yml]: Done.

File: EDSAnalytics/EDSTest/README.md (hunk 6 above)
Comment (gaholt) [reply EDSAnalytics/EDSTest/README.md]: Done.

File: EDSAnalytics/EDSTest/README.md (hunk 8 above)
Comment (gaholt) [reply EDSAnalytics/EDSTest/README.md]: Fixed.

File: EDSAnalytics/EDSTest/README.md (hunk 7 above)
Comment (gaholt) [reply EDSAnalytics/EDSTest/README.md]: Done.

**Review by dcbroad3 — APPROVED** [review]

**Pull request merged**